app/frontend/chat_interface.py

```diff
@@ -72,6 +72,14 @@ def load_styles():
 if "session_counter" not in st.session_state:
     st.session_state.session_counter = 1
 
+# Weather toggle persisted per-session (not personally identifiable)
+if "use_weather" not in st.session_state:
+    st.session_state.use_weather = True
+
+# Simple holder for last derived weather summary block
+if "weather_summary" not in st.session_state:
+    st.session_state.weather_summary = None
```

Comment:
This approach may lead to an inconsistent state if multiple users are served. Store user/session-specific state in a more isolated structure.
```
st.session_state['use_weather'] = st.checkbox("Enable weather-aware recommendations", value=st.session_state.get('use_weather', True))
```